doc/design/paths-endpoint-webdav.md

```diff
@@ -0,0 +1,21 @@
+# Discussion on metadata needed for dandidav
+
+Submitted as a PR to avoid spagetti thread discussion on https://github.com/dandi/dandi-archive/issues/1837#issuecomment-1921864949
+
+
+@jwodder listed needed properties and metadata fields:
+
+* Asset properties used by dandidav:
+    * `asset_id` (PRESENT)
+    * `blob_id`
+    * `zarr_id`
```

Comment:
I guess `zarr_id` is needed for traversal etc? or could contentUrl version be used? (probably we do not want to rely on parsing it).

So it feels that we might ask for asset record to include `blob_id` (although not sure what for yet exactly) and `zarr_id`, hence

```suggestion
    * `zarr_id`  (DESIRED)
```

Reply:
`dandidav` does parse S3 `contentUrl`s, as that's necessary in order to discover the name of the S3 bucket without hardcoding it or requiring it to be passed on the command line.

Reply:
so in principle we can just parse out bucket, zarr_id and blob_id from contenturl, correct?

Reply:
Yes, but I'd rather dandidav not have to make too many assumptions about how our S3 URLs are structured.

Reply:
agree. We would only need to be able to construct zarr "http s3" URLs from manifests so it would need to know that , right?
For blobs -- contentUrl should be good enough.

Reply:
Note that asset metadata lists two `contentUrl`s, an unsigned S3 URL and an API download URL of the form `https://api.dandiarchive.org/api/assets/{asset_id}/download/`.

* For Zarr assets, `dandidav` needs the S3 URL in order to know how to query the Zarr's entries from S3.

* However, for blob assets, `dandidav` currently uses the API download URL, as it provides a better UX.  Specifically, if `dandidav` were to redirect `GET` requests for a blob to the blob's unsigned S3 URL, then the browser would save the response to a file named after the last component of the S3 URL, which is (always?) the blob ID.  In order to get the downloaded files to be named after the filename portion of the asset's path instead, a signed S3 URL with a Content-Disposition has to be used, and that is acquired by redirecting to the API download URL, which in turn redirects to such a signed S3 URL.